src/main/java/appeng/core/AELog.java

```diff
@@ -19,93 +19,374 @@
+	private static final String CLIENT_SUFFIX = "C";
+
+	private static final String BLOCK_UPDATE = "Block Update of %s @ ( %d, %d, %d )";
+
+	private static final String DEFAULT_EXCEPTION_MESSAGE = "Exception: ";
 
 	private AELog()
 	{
 	}
 
-	public static void warning( final String format, final Object... data )
+	/**
+	 * Returns a {@link Logger} logger suitable for the effective side (client/server).
+	 *
+	 * @return a suitable logger instance
+	 */
+	private static Logger getLogger()
+	{
+		final String loggerName = LOGGER_PREFIX + ( Platform.isServer() ? SERVER_SUFFIX : CLIENT_SUFFIX );
+		final Logger logger = LogManager.getLogger( loggerName );
+
+		return logger;
+	}
+
+	/**
+	 * Indicates of the global log is enabled or disabled.
+	 *
+	 * By default it is enabled.
+	 *
+	 * @return true when the log is enabled.
+	 */
+	public static boolean isLogEnabled()
 	{
-		log( Level.WARN, format, data );
+		return AEConfig.instance == null || AEConfig.instance.isFeatureEnabled( AEFeature.Logging );
 	}
 
-	private static void log( final Level level, final String format, final Object... data )
+	/**
+	 * Logs a formatted message with a specific log level.
+	 *
+	 * This uses {@link String#format(String, Object...)} as opposed to the {@link ParameterizedMessage} to allow a more
+	 * flexible formatting.
+	 *
+	 * The output can be globally disabled via the configuration file.
+	 *
+	 * @param level the intended level.
+	 * @param message the message to be formatted.
+	 * @param params the parameters used for {@link String#format(String, Object...)}.
+	 */
+	public static void log( @Nonnull final Level level, @Nonnull final String message, final Object... params )
 	{
-		if( AEConfig.instance == null || AEConfig.instance.isFeatureEnabled( AEFeature.Logging ) )
+		if( AELog.isLogEnabled() )
 		{
-			FMLRelaunchLog.log( "AE2:" + ( Platform.isServer() ? "S" : "C" ), level, format, data );
+			final String formattedMessage = String.format( message, params );
+			final Logger logger = getLogger();
+
+			logger.log( level, formattedMessage );
 		}
 	}
 
-	public static void grinder( final String o )
+	/**
+	 * Log an exception with a custom message formated via {@link String#format(String, Object...)}
+	 *
+	 * Similar to {@link AELog#log(Level, String, Object...)}.
+	 *
+	 * @see AELog#log(Level, String, Object...)
+	 *
+	 * @param level the intended level.
+	 * @param exception
+	 * @param message the message to be formatted.
+	 * @param params the parameters used for {@link String#format(String, Object...)}.
+	 */
+	public static void log( @Nonnull final Level level, @Nonnull final Throwable exception, @Nonnull String message, final Object... params )
 	{
-		if( AEConfig.instance.isFeatureEnabled( AEFeature.GrinderLogging ) )
+		if( AELog.isLogEnabled() )
 		{
-			log( Level.DEBUG, "grinder: " + o );
+			final String formattedMessage = String.format( message, params );
+			final Logger logger = getLogger();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Make the "logger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.orod.org/coding_rules#rule_key=squid%3AS1312)